### ISSUE TITLE
Unable to download files greater than 2GB

### ISSUE THREAD

**Issue by Skyggedans** [fluttercommunity/flutter_downloader#370]
There's an integer truncation while getting the Content-Length header value on Android which makes it impossible to download files greater than 2GB.
Please consider the following fix:
```diff
@@ -298,7 +298,7 @@ public class DownloadWorker extends Worker implements MethodChannel.MethodCallHa

             if ((responseCode == HttpURLConnection.HTTP_OK || (isResume && responseCode == HttpURLConnection.HTTP_PARTIAL)) && !isStopped()) {
                 String contentType = httpConn.getContentType();
-                int contentLength = httpConn.getContentLength();
+                long contentLength = httpConn.getContentLengthLong();
                 log("Content-Type = " + contentType);
                 log("Content-Length = " + contentLength);
```

**Comment by hnvn** [comment]
Thanks for your feedback, could you make it as a PR?

**Comment by Skyggedans** [comment]
PR #371 created.

**Comment by bartekpacia** [comment]
fixed in #371
